server/src/test/java/org/elasticsearch/cluster/metadata/DateMathExpressionResolverTests.java

```diff
@@ -70,6 +71,27 @@ public void testExpression() throws Exception {
         assertThat(result.get(2), equalTo("logstash-" + formatDate("uuuu.MM.dd", dateFromMillis(context.getStartTime()))));
     }
 
+    public void testExpressionWithWildcardAndExclusions() {
+        List<String> indexExpressions = Arrays.asList(
+            "<-before-inner-{now}>",
+            "-<before-outer-{now}>",
+            "<wild*card-{now}*>",
+            "<-after-inner-{now}>",
+            "-<after-outer-{now}>"
+        );
+        List<String> result = DateMathExpressionResolver.resolve(context, indexExpressions);
+        assertThat(
+            result,
+            Matchers.contains(
+                equalTo("-before-inner-" + formatDate("uuuu.MM.dd", dateFromMillis(context.getStartTime()))),
+                equalTo("-<before-outer-{now}>"), // doesn't evaluate because it doesn't start with "<" and it is not an exclusion
```

Comment:
Are the other exclusions without date math dependent on the order in which they are defined ? i.e. does `-b,c*,-d` care about the order in which they are defined ? 

I would intuitively think that `-<index-{now/d}>,*` would behaves the same as `*,-<index-{now/d}>`

Reply:
> Are the other exclusions without date math dependent on the order in which they are defined ? i.e. does -b,c*,-d care about the order in which they are defined ?

Yes, names that start with `-` are not considered exclusions if there's no previous wildcard, i.e.
`-b,c*,-d` would error as `-b "resource_not_found"`.

> I would intuitively think that -<index-{now/d}>,* would behaves the same as *,-<index-{now/d}>

Maybe it would be more intuitive, but names starting with `-` before a wildcard are not considered exclusions, and because this is not an exclusion it is also not a datemath, because it doesn't start with `<`, so we have to treat this as a malformed index name that throws a resource not found.
Thats how I view it at least.

Reply:
thanks for the additional explanation. ++ that the exclusion behavior is consistent